modules/core/src/main/scala/org/tessellation/l0/domain/snapshot/GlobalSnapshot.scala

```diff
@@ -0,0 +1,19 @@
+package org.tessellation.l0.domain.snapshot
+
+import cats.data.NonEmptySet
+
+import org.tessellation.dag.domain.block.DAGBlock
+import org.tessellation.kernel.StateChannelSnapshot
+import org.tessellation.schema.height.Height
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.security.signature.Signed
+
+import io.estatico.newtype.ops._
+
+case class GlobalSnapshot(
+  blocks: Set[Signed[DAGBlock]],
```

Comment:
I wonder if we will be actually creating snapshots with no DAGBlocks 🤔

Reply:
I guess we will. That's why it is not `NonEmptySet`. I guess the types will change when we implement the trigger so I wouldn't worry about it now